go/vt/sqlparser/set_normalizer.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqlparser
+
+import (
+	"strings"
+
+	"vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+type setNormalizer struct {
+	err error
+}
+
+func (n *setNormalizer) rewriteSetComingUp(cursor *Cursor) bool {
+	set, ok := cursor.node.(*Set)
+	if ok {
+		for i, expr := range set.Exprs {
+			exp, err := n.normalizeSetExpr(expr)
+			if err != nil {
+				n.err = err
+				return false
```

Comment:
Have you tested this code path? I see a panic in rewriter.go if this returns false.

Reply:
Yes, it is tested in the unit tests. All cases where we expect errors pass this path. The magic is here: https://github.com/vitessio/vitess/blob/master/go/vt/sqlparser/rewriter_api.go#L40